src/index.spec.ts

```diff
@@ -1,12 +1,203 @@
+import cases from 'jest-in-case';
+import fetch, { Response } from 'node-fetch';
 import {
   isInteger,
   toLowerCase,
   removeDuplicatesFromArray,
   createRandomProduct,
   getStarWarsPlanets,
-  createProduct 
+  createProduct,
 } from './index';
 
-test('Testing time!', () => {
-  console.log('welcome');
+jest.mock('node-fetch', () => {
+  const module = jest.requireActual('node-fetch');
+  return {
+    __esModule: true,
+    ...module,
+    default: jest.fn(module.default),
+  };
+});
+
+function casify(subcases: {}) {
+  return Object.entries(subcases).map(([testName, testValue]) => {
+    return {
+      name: `${testValue} - ${testName}`,
+      testValue,
+    };
+  });
+}
+
+function functionResultOrError(aFunction: Function, argument) {
+  try {
+    return aFunction(argument);
+  } catch (e) {
+    return e;
+  }
+}
+
+describe(`isInteger`, () => {
+  cases(
+    `returns true if integer`,
+    (subcase) => expect(isInteger(subcase.testValue)).toBe(true),
+    casify({
+      postive: 3,
+      negative: -3,
+      '.0 decimal': 3.0,
+    }),
+  );
+
+  cases(
+    `returns false if not integer`,
+    (subcase) => expect(isInteger(subcase.testValue)).toBe(false),
+    casify({
+      string: '3',
+      decimal: 3.1,
+    }),
+  );
+});
+
+describe(`toLowerCase`, () => {
+  test(`requires non empty string`, () => {
+    const str = '';
+    expect(toLowerCase(str)).toMatchInlineSnapshot(`"Please provide a string"`);
+  });
+
+  test(`returns lowercase if string has length`, () => {
+    const str = 'A Message';
+    expect(toLowerCase(str)).toBe('a message');
+  });
+});
+
+describe(`removeDuplicatesFromArray`, () => {
+  test(`returns error if not an array`, () => {
+    const array = 3;
+    const error = functionResultOrError(removeDuplicatesFromArray, array);
+    expect(error).toMatchInlineSnapshot(
+      `[Error: please provide an array of numbers or strings]`,
+    );
+  });
+
+  test(`returns identity if just one element`, () => {
+    const stringArray = ['a'];
+    const numberArray = [1];
+    expect(removeDuplicatesFromArray(stringArray)).toEqual(stringArray);
+    expect(removeDuplicatesFromArray(numberArray)).toEqual(numberArray);
+  });
+
+  cases(
+    `returns array without duplicates if many elements`,
+    (subcase) => {
+      if (typeof subcase.testValue[0] === 'string') {
+        expect(removeDuplicatesFromArray(subcase.testValue)).toEqual([
+          'a',
+          'b',
+          'c',
+        ]);
+      } else if (typeof subcase.testValue[0] === 'number') {
+        expect(removeDuplicatesFromArray(subcase.testValue)).toEqual([1, 2, 3]);
+      }
+    },
+    casify({
+      stringUniques: ['a', 'b', 'c'],
+      stringDuplicated: ['a', 'a', 'b', 'b', 'c', 'c'],
+      numberUniques: [1, 2, 3],
+      numberDuplicated: [1, 1, 2, 2, 3, 3],
+    }),
+  );
+});
+
+describe(`createProduct`, () => {
+  test(`throws error if not a valid schema product`, () => {
```

Comment:
Remove unnecessary backticks. Try to configure your linter to help you to detect this kind of code smells.